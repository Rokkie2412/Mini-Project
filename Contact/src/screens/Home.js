import React from "react";
import {Text,TextInput,View,Pressable} from "react-native"
import styles from '../style/HomeStyle'
import IonIcon from 'react-native-vector-icons/Ionicons'

const Home = ({navigation}) =>{
    return(
        <View style={styles.mainContainer}>
            <View style={styles.Header}>
                <Text style={styles.HeaderText}>Contact List</Text>
            </View>
            <View style={styles.searchbar}>
                <IonIcon name="search" style={styles.searchIcon} size={20}/>
                <TextInput
                    style={styles.TextInputSearch}
                    placeholder="Search Contact"
                />
                <IonIcon name="close" style={styles.clearIcon} size={18}/>
            </View>
            <View style={styles.bodyContainer}>
                <Text>There is no current contact here</Text>
            </View>
            <View style={styles.AddContainer}>
                <Pressable style={styles.addSection}
                onPress={()=>{
                    navigation.navigate('Add')
                }}>
                    <IonIcon style={styles.addIcon} name="add" size={35}/>
                </Pressable>
            </View>
        </View>
    )
}


export default Home