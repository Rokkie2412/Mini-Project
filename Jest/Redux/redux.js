export const ADD = 'ADD'
export const DELETE = 'DELETE'

export const addArray = (data) =>({
    type:ADD,
    payload:data,
})

export const deleteFromArray = (title) => ({
    type:DELETE,
    payload:title,
})

let initialState = {
    arr:[]
}

export const mainReducer = (state=initialState,action) => {
    switch(action.type){
        case ADD:
            return{
                ...state,arr:[...state.arr,action.payload]
            }
        case DELETE:
            return{
                ...state,arr:state.arr.filter(arr => arr.id !== action.payload)
            }
        default:
            return state
    }
}

export default mainReducer