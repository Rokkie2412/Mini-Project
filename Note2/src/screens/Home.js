import React, {Component} from 'react'
import {View,Text, TouchableOpacity,StyleSheet,ScrollView} from 'react-native'
import {FAB} from 'react-native-elements'

class Home extends Component {
    state = { 
        objectData:{

        }
     }
    render() {
        return (
          <View style={styles.mainContainer}>
              <View>
                  <Text style={styles.title}>Notes</Text>
              </View>
              <View style={styles.body}>
                  <Text style={{color:'#333',fontSize:14}}>You don't have any Notes</Text>
              </View>
              <View style={{alignItems:'flex-end'}}>
                <TouchableOpacity style={styles.FAB} onPress={()=> this.props.navigation.navigate('AddNotes')}>
                    <Text style={{color:'#fff',fontSize:26}}>+</Text>
                </TouchableOpacity>
              </View>  
          </View>  
        );
    }
}

const styles = StyleSheet.create({
    mainContainer:{
        paddingVertical:30,
        paddingHorizontal:30,
        backgroundColor: '#fff',
        flex:1,

    },
    body:{
        flex:1,
        justifyContent:'center',
        alignItems:'center',
    },
    title:{
        fontSize:26,
        color:'#333',
        fontWeight:'bold',
        alignContent:'center'
    },
    FAB:{
        alignContent:'center',
        alignItems:'center',
        backgroundColor:'#333',
        fontSize:30,
        color:'#fff',
        padding:12, 
        borderRadius:100,
        width:60,
    }
})

export default Home;