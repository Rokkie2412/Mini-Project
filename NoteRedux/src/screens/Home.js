import React from 'react'
import {View,Text,StyleSheet,TouchableOpacity} from 'react-native'

const Home = (props) => {
    return(
        <View style={styles.container}>
            <View style={styles.header}>
                <Text style={styles.haderText}>Note App</Text>
            </View>
            <View style={styles.body}>
                <Text>
                    There is no note right now
                </Text>
            </View>
            <View style={styles.button}>
                <TouchableOpacity
                onPress={()=>{ 
                    props.navigation.navigate('AddNotes')
                }}
                >
                    <Text style={styles.buttonText}>+</Text>
                </TouchableOpacity>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    container:{
        backgroundColor:'#F1FFFF',
        flex:1
    },
    header:{
        backgroundColor:"#00ADB5",
        borderRadius:50,
        width:130,
        justifyContent:'center',
        alignItems:'center',
        marginLeft:20,
        marginVertical:15
    },
    haderText:{
        fontSize:22,
        padding:10,
        color:"#222831",
        fontWeight:'bold'
    },
    body:{
        alignItems:'center',
        alignSelf:'center',
        flex:1,
        alignContent:'center',
        justifyContent:'center'
    },
    textBody:{
        fontSize:17,
    },
    button:{
        flexDirection:"row",
        backgroundColor:'#00ADB5',
        height:60,
        marginVertical:20,
        justifyContent:'flex-end',
        marginLeft:320,
        justifyContent:'center',
        borderRadius:100,
        alignItems:'center',
        marginRight:10
        
    },
    buttonText:{
        fontSize:30,
        paddingBottom:4
    }
})

export default Home