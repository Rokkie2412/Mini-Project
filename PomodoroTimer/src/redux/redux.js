
/**
 * @format
 * @flow
 */
export const EXCHAHNGETOSECONDS : string = 'EXCHANGETOSECOND'
export const EXCHAHNGETOSECONDBREAK :string = 'EXCHANGETOSECONDBREAK'
export const EXCHAHNGETOSECONDLONGBREAK :string = 'EXCHANGETOSECONDLONGBREAK'
export const GETDATA:string = 'GETDATA'

export const exchangetosecwork = (hour:string,minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDS,
    payloadhourwork : hour,
    payloadminutework : minute,
    payloadsecondwork : second
})

export const exchangetosecBreak = (minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDBREAK,
    payloadminutebreak : minute,
    payloadsecondbreak : second
})

export const exchangetosecLongBreak = (minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDLONGBREAK,
    payloadminutelongbreak : minute,
    payloadsecondlongbreak : second
})

let initialState : Object  ={
    resultWorkinSecond : 0,
    resultBreakinSecond : 0,
    resultLongBreakinSecond : 0
}

export const mainReducer = (state:Object=initialState, action:function):Function =>{
    switch (action.type) {
        case EXCHAHNGETOSECONDS:
            
            const hourTosecond : number = action.payloadhourwork * 3600
            const minuteTosecond : number = action.payloadminutework * 60
            const res : number = hourTosecond+minuteTosecond+action.payloadsecondwork
            return {
                ...state,
                resultWorkinSecond:res
            }
        case EXCHAHNGETOSECONDBREAK:
            const minuteTosecondBreak : number= action.payloadminutebreak * 60
            const resBreak : number = minuteTosecondBreak+action.payloadsecondbreak
            return {
                ...state,
                resultBreakinSecond:resBreak
            }
        case EXCHAHNGETOSECONDLONGBREAK:
            const minuteToSecondLongBreak : number= action.payloadminutelongbreak * 60
            const resLongBreak : number = minuteToSecondLongBreak+action.payloadsecondlongbreak
            return{
                ...state,
                resultLongBreakinSecond:resLongBreak
            }
        default:
            return state
    }
}

export default mainReducer