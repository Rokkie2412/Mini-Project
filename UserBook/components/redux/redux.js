export const ADD_CONTACT = 'ADD_CONTACT'
export const SHOW_CONTACT = 'SHOW_CONTACT'
export const EDITCONTACT = 'EDITCONTACT'

const ApiUrl = 'https://simple-contact-crud.herokuapp.com/contact'


export const addContact = (firstName,lastName,age,image) =>({
    type:ADD_CONTACT,
    payloadFirstname : firstName,
    payloadLastName : lastName,
    payAge : age,
    payloadImage : image
})

export const editContact = (itemId,firstName,lastName,age,image) => ({
    type:EDITCONTACT,
    payloadID : itemId,
    payloadFirstname : firstName,
    payloadLastName : lastName,
    payAge : age,
    payloadImage : image
})

let initialState = {
    arr:[],
}

export const mainReducer = (state=initialState, action) => {
    switch (action.type) {
        case ADD_CONTACT:
            const postContact = () =>{
                fetch('https://simple-contact-crud.herokuapp.com/contact',{
                    method:'POST',
                    body: JSON.stringify({
                        firstName : action.payloadFirstname,
                        lastName : action.payloadLastName,
                        age : action.payAge,
                        photo : action.payloadImage
                    }),
                    headers:{
                        'Content-Type' : 'application/json',
                    }
                }).then(Response=>Response.json()).then(json=>console.log(json))
            }
            return(postContact())
        case EDITCONTACT :
            const editContact = () =>{
                const tempid = action.payloadID 
                
                fetch(`https://simple-contact-crud.herokuapp.com/contact/${tempid}`,{
                    method:'PUT',
                    body: JSON.stringify({
                        firstName : action.payloadFirstname,
                        lastName : action.payloadLastName,
                        age : action.payAge,
                        photo : action.payloadImage
                    }),
                    headers:{
                        'Content-Type' : 'application/json',
                    }
                 }).then(Response=>Response.json()).then(json=>console.log(json))
            }
            return(editContact())
        default:
            return state
    }
}