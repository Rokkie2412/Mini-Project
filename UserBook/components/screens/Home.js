import React,{useState, useEffect} from 'react'
import {View,Text,StyleSheet,Pressable,FlatList,TouchableOpacity,Image} from 'react-native'
import SearchBar from './SearchBar'
import { useSelector} from "react-redux";
import { showMessage } from 'react-native-flash-message';
import Ion from 'react-native-vector-icons/Ionicons'
import NetInfo from "@react-native-community/netinfo";
import RNRestart from 'react-native-restart';

const Home = ({navigation}) => {
    const [search,setSearch] = useState('')
    const [result,setResult] = useState([])
    const [Filterresult,setFilterResult] = useState([])
    let dataArray = useSelector(state=>state.arr)
    dataArray = Filterresult
    
    const intervalSet = () => {
        setInterval(()=>{
        checkInternet()
        clearInterval(10000)
    },10000)
    }

    const IntervalCheck = () => {
        const interval = intervalSet()
        return clearInterval(interval)
    }

    useEffect(()=>{
        getData()
        checkInternet()
        const unsubscribe = navigation.addListener('focus',()=>{
            getData()
        })
        return unsubscribe
    },[navigation])

    const getData = () => {
        fetch('https://simple-contact-crud.herokuapp.com/contact')
        .then((response)=>response.json())
        .then((json)=>{
            setResult(json.data)
            setFilterResult(json.data)
        })
        .catch((error)=>{
            showMessage({
                message:'Error Happened',
                description:`${error} is happened, please contact your admin`,
                floating:true,
                type:'danger',
                icon:'danger',
                animationDuration:'800',
                autoHide:false,
                onPress:()=>{
                    RNRestart.Restart()
                }
            })
        })
    }


    const checkInternet = () =>{
        NetInfo.fetch('https://simple-contact-crud.herokuapp.com/contact').then(networkState=>{
        // console.log("Is connected? - ", networkState.isConnected);
        if (networkState.isConnected !== true) {
            showMessage({
                message:'Not connected to server',
                description:'Connection Error Check Your Internet Connection, Press to restart app',
                floating:true,
                type:'warning',
                icon:'warning',
                animationDuration:600,
                autoHide:false,
                onPress: () => {
                    RNRestart.Restart()
                    }
                })
            }
        })
        
    }

    const FilterSearch = (text) => {
        if (text){
            const newData = result.filter((item)=>{
                const itemData = item.firstName ? item.firstName.toUpperCase() : ''.toUpperCase()
                const textData = text.toUpperCase()
                return itemData.indexOf(textData) > -1
            })
            setFilterResult(newData)
            setSearch(text)
        }else{
            setFilterResult(result)
            setSearch(text)
        }
        
    }

    IntervalCheck()
    
    return(
        <View style={styles.mainContainer}>
            <View style={styles.header}>
                <Text style={styles.headerTitle}>Contacts</Text>
            </View>

            <View>
                <SearchBar search={search} setSearch={(text)=>FilterSearch(text)}/>
            </View>
            
            <View style={styles.flatlistView}>
                <FlatList
                    showsVerticalScrollIndicator={false}
                    data={dataArray.sort((a,b)=>a.firstName.localeCompare(b.firstName))}
                    keyExtractor={({id},index)=>id}
                    renderItem={({item})=>{
                        return(
                            <View style={styles.flatcontainer}>
                                
                                    <TouchableOpacity onPress={()=>{
                                        navigation.navigate('EditContact',
                                            {itemID : item.id,
                                            firstName: item.firstName,
                                            lastName : item.lastName,
                                            age:item.age,
                                            photo:item.photo})
                                    }}>
                                    <View style={styles.contactList}>
                                        {item.photo == 'N/A' ? <Ion name='person' style={styles.peopleIcon}/> :
                                        <Image style={styles.imagePhoto} source={{uri:`${item.photo}`}}/>
                                        }
                                        <Text style={styles.contactName}>{item.firstName} {item.lastName}</Text>
                                    </View>
                                    </TouchableOpacity>
                                
                            </View>
                        )
                    }}
                />
            </View>
            <View style={styles.buttonContainer}>
                <Pressable style={{alignItems:'flex-end'}} 
                onPress={()=>navigation.navigate('AddContact')}
                >
                    <Ion name='add' style={styles.addButton}/>
                </Pressable>
            </View>
            
        </View>
    )
}

const styles = StyleSheet.create({ 
    mainContainer:{
        flex:1,
        backgroundColor:'#FFF'

    },
    header:{
        flexDirection:'row',
        justifyContent:'center',
    },
    headerTitle:{
        fontSize:24,
        color:'black',
        fontWeight:'500',
        marginTop:5,
    },
    buttonContainer:{
        
        // borderWidth:1,
        // borderColor:'black',
        // height:600,
    },
    addButton:{
        position:'absolute',
        fontSize:40,
        backgroundColor:'#00ae78',
        color:'#f8ffff',
        padding:10,
        borderRadius:50,
        bottom:20,
        right:13,
    },
    contactList:{
        flexDirection:'row',
        marginHorizontal:25,
        marginLeft:18,
        marginVertical:13,
    },
    contactName:{
        fontSize:20,
        color:'#2C3333',
        fontWeight:'500',
        marginLeft:15,
        textAlign:'center',
        alignSelf:"center"
    },
    peopleIcon:{
        fontSize:23,
        color:'#f8ffff',
        padding:10,
        backgroundColor:'#dee1e6',
        borderRadius:50,
    },
    imagePhoto:{
        width:47,
        height:47,
        borderRadius:50,
    },
    flatcontainer:{
        justifyContent:'center',
        flexGrow: 0
    },
    flatlistView:{
        flex:1,
    }
})

export default Home