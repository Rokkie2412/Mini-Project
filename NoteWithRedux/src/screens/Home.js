import React,{useState} from 'react'
import {View,Text,StyleSheet,TouchableOpacity,FlatList,Alert, Modal, Pressable,TextInput} from 'react-native'
import { useSelector,useDispatch } from 'react-redux'
import { Entypo } from '@expo/vector-icons'; 
import { deleteFromArray,EditNote,addArray } from '../redux/reduxs';
import { AntDesign } from '@expo/vector-icons'; 
const Home = ({navigation}) => {

    const [modal,setModal] = useState(false)
    let object={}
    const [TitleModal,setTitleModal] = useState('')
    const [BodyModal,setBodyModal] = useState('')
    const data = useSelector(state => state)
    const dispatch = useDispatch()
    let tempTitleModal = ''
    let tempBodyModal = ''
    const removeNote = (title) => {
        dispatch(deleteFromArray(title))
    }

    // const editNote = (title) => {
    //     dispatch(EditNote(title))
    // }

    const editFilter = (title,body) => {
        data.filter((val)=>{
            if(val.title === title)
                setTitleModal(val.title)
                tempTitleModal = val.title
                
        })
        data.filter((val)=>{
            if(val.body === body)
                setBodyModal(val.body)
                tempBodyModal = val.body
                
        })
    }

    const addNote = () => {
        object.title=TitleModal
        object.body = BodyModal
        dispatch(addArray(object))
        console.log(data);
    }


    return(
        <View style={styles.container}>

            {/* //MODAL VIEW */}

            {/* MAIN VIEW */}
            <View style={styles.header}>
                <Text style={styles.haderText}>Note App</Text>
            </View>
            {/* FLATLIST */}
            <View style={styles.body}>
                <FlatList
                data={data}
                keyExtractor={(data)=>data.title}
                renderItem={({item})=>{
                    return(
                        <View>
                            <Modal
                            animationType='slide'
                            transparent={true}
                            visible={modal}
                            onRequestClose={()=>{
                                setModal(!modal)
                            }}
                            >
                                <View style={styles.modalContainer}>
                                    <View style={styles.modalnav}>
                                        <TouchableOpacity 
                                        onPress={()=>{
                                                    
                                                    addNote()
                                                    removeNote(item.title)
                                                    setModal(false)
                                                }}
                                        >
                                                <AntDesign style={styles.modalBackIcon} name="back" size={24} color="black" /></TouchableOpacity>
                                        <Text style={styles.ModalTitle}>Edit Note</Text>
                                    </View>
                                    <View style={styles.modalInput}>
                                        <TextInput
                                        autoCapitalize='none'
                                        autoCorrect={false}
                                        placeholder='Edit Title Here'
                                        style={styles.editTitle}
                                        value={TitleModal}
                                        onChangeText={(newTitle)=>{
                                            setTitleModal(newTitle)
                                        }}
                                        />

                                        <TextInput
                                        autoCapitalize='none'
                                        autoCorrect={false}
                                        placeholder='Edit Body Here'
                                        style={styles.editBody}
                                        multiline={true}
                                        numberOfLines={15}
                                        value={BodyModal}
                                        onChangeText={(newBody)=>{
                                            setBodyModal(newBody)
                                        }}
                                        />

                                    </View>
                                </View>
                            </Modal>
                            <Text>
                            <TouchableOpacity
                                onPress={()=>{
                                    editFilter(item.title,item.body)
                                    setModal(true)
                                }}
                                onLongPress={()=>{
                                    removeNote(item.title)
                                }}
                            >
                                <View style={styles.flatview}>
                                    <Text style={styles.flattexttitle}>{item.title}</Text>
                                    <Text style={styles.flattextbody}>{item.body}</Text>
                                </View>
                            </TouchableOpacity>
                        </Text>
                        </View>
                    )
                }}
                />
            </View>
            <View style={styles.touch}>
                <TouchableOpacity onPress={()=>{
                    setModal(false)
                    navigation.navigate('AddNotes')
                    console.log(data);
                }}>

                    <View style={styles.button}>
                        <Text style={styles.buttonText}>+</Text>
                    </View>
                </TouchableOpacity>
            </View>

        </View>
    )
}

const styles = StyleSheet.create({
    container:{
        backgroundColor:'#F1FFFF',
        flex:1,
        paddingTop:22
    },
    header:{
        backgroundColor:"#00ADB5",
        borderRadius:50,
        width:130,
        justifyContent:'center',
        alignItems:'center',
        marginLeft:20,
        marginVertical:15
    },
    haderText:{
        fontSize:22,
        padding:10,
        color:"#222831",
        fontWeight:'bold'
    },
    body:{
        flex:1,

    },
    textBody:{
        fontSize:17,
    },
    button:{
        backgroundColor:'#00ADB5',
        height:70,
        marginVertical:20,
        alignItems:'center',
        justifyContent:'center',
        borderRadius:50,
        width:70,
        marginHorizontal:13
    },
    buttonText:{
        fontSize:30,
        paddingBottom:4,
        fontWeight:'bold',
        color:'#EEEEEE'
    },
    touch:{
        flexDirection:"row",
        justifyContent:'flex-end',
    },
    flatview:{
        marginHorizontal:30,
        backgroundColor:'#00ADB5',
        width:350,
        maxHeight:'90%',
        borderRadius:50,
        paddingVertical:10,
        paddingHorizontal:20,
        marginVertical:5
    },
    flattexttitle:{
        color: '#EEEEEE',
        fontSize:18,
        fontWeight:'500',
        padding:5,
        paddingLeft:10
    },
    flattextbody:{
        color:'#EEEEEE',
        fontSize:14,
        fontWeight:'300',
        paddingLeft:8
    },
    modalContainer:{
        flex:1,
        margin: 20,
        marginVertical:30,
        borderRadius:30,
        backgroundColor:'#A6E3E9'
    },
    modalnav:{
        padding:10,
        flexDirection:'row'
    },
    modalBackIcon:{
        fontSize:26,
        padding:10,
        color:'#062C30'
    },
    ModalTitle:{
        paddingHorizontal:20,
        paddingVertical:10,
        backgroundColor:'#CBF1F5',
        borderRadius:40,
        fontSize:20,
        color:'#062C30',
        fontWeight:'700'
    },
    editTitle:{
        fontSize:18,
        marginVertical:10,
        marginHorizontal:22,
        borderBottomWidth:1,
        padding:4,
        borderColor:'#00ADB5'
    },
    editBody:{
        marginVertical:10,
        marginHorizontal:22,
        textAlignVertical:'top',
        fontSize:16,
    }
})

export default Home