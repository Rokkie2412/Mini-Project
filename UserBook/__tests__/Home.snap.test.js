import React from "react";
import Home from "../components/screens/Home";
import { Provider } from "react-redux";
import { mainReducer } from "../components/redux/redux";
import { createStore } from "redux";
import { create } from "react-test-renderer";

jest.mock(
  'react-native-vector-icons/Ionicons',
  () => 'MockedIonicons',
);

jest.mock(
    'react-native-flash-message',
    ()=> 'Mockedreact-native-flash-message'
)


jest.mock("@react-native-community/netinfo", () => ({
  fetch: () => Promise.resolve(jest.fn()),
}));

const navigation={
    navigate:jest.fn()
}

const store = createStore(mainReducer,{status:'default status'})
const tree = create(
    <Provider store={store}>
        <Home navigation={navigation}/>
    </Provider>
)

test('Snap test Home pages',()=>{
    expect(tree).toMatchSnapshot()
})

test('navigation to Add Contact',()=>{
    const button = tree.root.findByProps({testID:'myButton'}).props
    button.onPress()
    expect(navigation.navigate).toBeCalledWith('AddContact')
})