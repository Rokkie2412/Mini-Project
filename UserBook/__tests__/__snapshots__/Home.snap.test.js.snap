// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snap test Home pages 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 24,
          "fontWeight": "500",
          "marginTop": 5,
        }
      }
    >
      Contacts
    </Text>
  </View>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "lightgrey",
          "borderRadius": 60,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "center",
          "marginHorizontal": 20,
          "marginVertical": 10,
        }
      }
    >
      <MockedIonicons
        name="search"
        size={28}
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "marginHorizontal": 10,
          }
        }
      />
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="Search contact"
        style={
          Object {
            "flex": 1,
            "fontSize": 17,
          }
        }
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      data={Array []}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View />
    </RCTScrollView>
  </View>
  <View
    style={Object {}}
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
        }
      }
      testID="myButton"
    >
      <MockedIonicons
        name="add"
        style={
          Object {
            "backgroundColor": "#00ae78",
            "borderRadius": 50,
            "bottom": 20,
            "color": "#f8ffff",
            "fontSize": 40,
            "padding": 10,
            "position": "absolute",
            "right": 13,
          }
        }
      />
    </View>
  </View>
</View>
`;
